src/shared/wizards/multiStepWizard.ts

```diff
@@ -12,17 +12,74 @@ import * as vscode from 'vscode'
 import * as picker from '../../shared/ui/picker'
 
 export interface WizardStep {
-    (): Thenable<WizardStep | undefined>
+    (): Thenable<Transition>
+}
+
+export interface Transition {
+    nextState: NextWizardState
+    nextStep?: WizardStep
+}
+
+export enum NextWizardState {
```

Comment:
the `WIZARD_` prefix in this PR is great for discoverability. Could also use "Wizard" prefix here without a loss in clarity.

```suggestion
export enum WizardNextState {
```